pkg/authentication/bootstrapauthenticator/bootstrap_test.go

```diff
@@ -0,0 +1,93 @@
+package bootstrapauthenticator
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestIsEnabled(t *testing.T) {
+	namespaceSystem := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: metav1.NamespaceSystem,
+		},
+	}
+	testCases := []struct {
+		name        string
+		kubeClient  *fake.Clientset
+		expectedErr string
```

Comment:
Before addressing any other comments - you don't use this field at all, do you need it, then? Or you should remove the `namespaceSystem` in one of the TCs and use it, although such a scenario does not sound very likely.

Also, have go default this field for you (it'll make it be an empty string)

Reply:
i added this field to test out any error cases but was unable to accomplish without lot of mocking. IMO, removing the `namespaceSystem` is the simplest we can do to test. Let me just add that testcase.